allen-cell-animated/nucmorph-colorizer
# refactor: React routing

Problem
=======
Takes the first steps towards #250 (adding the landing page) by adding React Router to the project and starting to restructure it. This will allow me to add the landing page later.

*Estimated review time: small, <15 minutes*

Solution
========
- Adds `react-router-dom` to the project.
- Renames `App` to `Viewer`.
- Moves some CSS out of `Viewer.module.css` and into `AppStyle` so that formatting is consistent across pages.
- Adds a new default error page to confirm that React Router is working.

## Type of change
* New feature (non-breaking change which adds functionality)


Steps to Verify:
----------------
1. Open the preview link: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-255/
2. The app should continue to operate as expected.
3. Open a preview for a bad URL. The error page should appear. Try clicking the link button to return to the homepage:  https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-255/#/bad

Note that GitHub Pages doesn't support single-page applications (SPAs), so we need to use hash-based routing to ensure that the extra URL pathing is handled locally by the client. (In a URL, anything after a `/#/` is handled client-side only.)

Screenshots (optional):
-----------------------
New error page: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-255/#/bad

![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/d4339a19-7b60-49fe-b76c-4fb089d6be95)
